src/ir/ty.rs

```diff
@@ -384,6 +385,9 @@ pub enum TypeKind {
     /// A floating point type.
     Float(FloatKind),
 
+    /// A complex floating point type. complex only exist for floating point.
```

Comment:
nit: I think `/// A complex floating point type.` is more than enough, since it implies "floating point", but I guess the second sentence doesn't harm. Please capitalize "Complex" if you keep it :)